numpy/_core/src/multiarray/mapping.c

```diff
@@ -1999,6 +1999,10 @@ array_assign_subscript(PyArrayObject *self, PyObject *ind, PyObject *op)
                 goto fail;
             }
 
+            if (solve_may_share_memory(self, tmp_arr, 1) == 1) {
+                tmp_arr = (PyArrayObject *)PyArray_NewCopy(tmp_arr, NPY_ANYORDER);
```

Comment:
```suggestion
                Py_SETREF(tmp_arr, PyArray_NewCopy(tmp_arr, NPY_ANYORDER));
```
You would leak the old `tmp_arr`